Sources/DeepLinking/Router.swift

```diff
@@ -9,7 +9,9 @@
 import UIKit
 
 public protocol RouteHandler {
-    func handle(route: URL, parameters: [String : String], queryItems: [URLQueryItem])
+    associatedtype T
+
+    func handle(route: URL, parameters: [String : String], queryItems: [URLQueryItem], completion: ((T) -> Void)?)
```

Comment:
Just a thought. Do you think it might be nice to assign a default value of `nil` to the `completion`? This way, the caller can completely miss that parameter if they don't want to use it?

EDIT: I guess the same question for the route function below too 🤔